internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+)
+
+const (
+	defaultSidecarPort = "80"
+)
+
+// convertSidecar converts the manifest sidecar configuration into a format parsable by the templates pkg.
+func convertSidecar(s manifest.Sidecar) ([]*template.SidecarOpts, error) {
```

Comment:
unrelated to this PR, but now looking at it it's a bit strange to have a `Sidecar` struct that has a `Sidecars` field :P

Reply:
I think it was done for consistency with how we unmarshal yaml in the `manifest` package; the top level structs for something like a `LoadBalancedWebService` all use the `inline` YAML keyword, which is how we end up with 

```golang
type LoadBalancedWebService struct {
    TaskConfig `yaml:",inline"`
    Sidecar `yaml:",inline"`
}

type Sidecar struct {
    Sidecars map[string]*SidecarConfig `yaml:"sidecars"`
}
```

It makes more sense when considering a struct like `TaskConfig` with multiple inlined fields, but does look odd when there's only one subfield.

Reply:
Yeah, makes sense. We did that because we wanted to attach methods to the `Sidecar` type so we couldn't just do:
```go
type LoadBalancedWebService struct {
    TaskConfig `yaml:",inline"`
    Sidecars map[string]*SidecarConfig `yaml:"sidecars"`
}
```

But with your refactor, the `manifest` pkg only populates these structs so we could now clean this up in the future :)